profiler/controllers.py

```diff
@@ -1,70 +1,172 @@
     * profiler_player_enabled
     * profiler_player_disabled
     """
 
-    @openerpweb.jsonrequest
-    def enable(self, request):
-        logger.info("Enabling")
+    @http.route(['/web/profiler/enable'], type='json', auth="user")
+    def enable(self):
+        _logger.info("Enabling")
         core.enabled = True
+        ProfilerController.begin_date = datetime.now().strftime(
+            "%Y-%m-%d %H:%M:%S")
         ProfilerController.player_state = 'profiler_player_enabled'
 
-    @openerpweb.jsonrequest
-    def disable(self, request):
-        logger.info("Disabling")
+    @http.route(['/web/profiler/disable'], type='json', auth="user")
+    def disable(self, **post):
+        _logger.info("Disabling")
         core.enabled = False
+        ProfilerController.end_date = datetime.now().strftime(
+            "%Y-%m-%d %H:%M:%S")
         ProfilerController.player_state = 'profiler_player_disabled'
 
-    @openerpweb.jsonrequest
-    def clear(self, request):
+    @http.route(['/web/profiler/clear'], type='json', auth="user")
+    # @http.jsonrequest
+    def clear(self, **post):
         core.profile.clear()
-        logger.info("Cleared stats")
+        _logger.info("Cleared stats")
         ProfilerController.player_state = 'profiler_player_clear'
+        ProfilerController.end_date = ''
+        ProfilerController.begin_date = ''
 
-    @openerpweb.httprequest
-    def dump(self, request, token):
+    # @http.httprequest
+    @http.route(['/web/profiler/dump'], type='http', auth="user")
+    def dump(self, token, **post):
         """Provide the stats as a file download.
 
         Uses a temporary file, because apparently there's no API to
         dump stats in a stream directly.
         """
-        handle, path = mkstemp(prefix='profiling')
-        core.profile.dump_stats(path)
-        stream = os.fdopen(handle)
-        os.unlink(path)  # TODO POSIX only ?
-        stream.seek(0)
-        filename = 'openerp_%s.stats' % datetime.now().isoformat()
-        # can't close the stream even in a context manager: it'll be needed
-        # after the return from this method, we'll let Python's GC do its job
-        return request.make_response(
-            stream,
-            headers=[
-                ('Content-Disposition',
-                 'attachment; filename="%s"' % filename),
-                ('Content-Type', 'application/octet-stream')
-            ], cookies={'fileToken': token})
-
-    @openerpweb.jsonrequest
-    def initial_state(self, request):
-        user = request.session.model('res.users')
+        exclude_fname = self.get_exclude_fname()
+        with tools.osutil.tempdir() as dump_dir:
+            ts = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+            filename = 'openerp_%s' % ts
+            stats_path = os.path.join(dump_dir, '%s.stats' % filename)
+            core.profile.dump_stats(stats_path)
+            _logger.info("Pstats Command:")
+            _logger.info(
```

Comment:
Use a dictionary with all params and use `**params` from method call and use `%(key1)s` from logger string in order to avoid use 2 times the same parameters.